grails-app/services/hpo/api/HpoLoincService.groovy

```diff
@@ -13,16 +14,16 @@ import org.monarchinitiative.phenol.ontology.data.TermId
  */
 class HpoLoincService {
 
-  Loinc2HpoFactory loinc2HpoFactory
+  Loinc2Hpo hpoLoinc
 
   Set<LoincEntry> searchByHpo(TermId query) {
-    Set<LoincId> loincIds = loinc2HpoFactory.reverseAnnotationMap().get(query)
-    Map<LoincId, LoincEntry> loincEntryMap = loinc2HpoFactory.loincEntryMap()
+    Set<LoincId> loincIds = hpoLoinc.getReverseAnnotationMap().get(query)
+    Map<LoincId, LoincEntry> loincEntryMap = hpoLoinc.getLoincEntryMap()
 
     Set<LoincEntry> result = new HashSet<>()
 
-    for (LoincId loincid : loincIds) {
```

Comment:
Get Groovy about it. Easy Closure pattern here. Note.. you cant use "break" inside the closure.